lib/elixir/lib/calendar.ex

```diff
@@ -293,6 +293,30 @@ defmodule Date do
     end
   end
 
+  @doc """
+  Compares two `Date` structs. Returns :gt if first date is later than the second
+  and :lt for vice versa. If the two dates are equal :eq is returned
+
+  ## Examples
+
+    iex> Date.compare(Date.from_erl!({2016, 4, 16}), Date.from_erl!({2016, 4, 28}))
+    :lt
+  """
+  @spec compare(Date.t, Date.t) :: :lt | :eq | :gt
+  def compare(%Date{} = date_1, %Date{} = date_2) do
+    do_compare(to_erl(date_1), to_erl(date_2))
+  end
+
+  ## Helpers
+
+  defp do_compare(date1, date2) do
+    cond do
+      date1 > date2 -> :gt
+      date1 < date2 -> :lt
+      true -> :eq
+    end
```

Comment:
This can be handled as guard clauses on a function. I believe it will offer better performance, than `cond`.